druid/examples/list.rs

```diff
@@ -85,39 +119,49 @@ fn ui_builder() -> impl Widget<AppData> {
 
     // Build a list with shared data
     lists.add_flex_child(
-        Scroll::new(
-            List::new(|| {
+        Scroll::new(List::new(|| {
+            fn label(text_color: Color) -> impl Widget<(AppData, usize)> {
                 Flex::row()
                     .with_child(
-                        Label::new(|(_, item): &(Vector<u32>, u32), _env: &_| {
+                        Label::new(|(_, item): &(AppData, usize), _env: &_| {
                             format!("List item #{}", item)
                         })
-                        .align_vertical(UnitPoint::LEFT),
+                        .with_text_color(text_color)
+                        .align_vertical(UnitPoint::LEFT)
+                        .on_click(
+                            |_ctx, (shared, item): &mut (AppData, usize), _env| {
+                                shared.toggle_select(*item);
+                            },
+                        ),
                     )
                     .with_flex_spacer(1.0)
                     .with_child(
                         Button::new("Delete")
-                            .on_click(|_ctx, (shared, item): &mut (Vector<u32>, u32), _env| {
-                                // We have access to both child's data and shared data.
+                            .on_click(|_ctx, (shared, item): &mut (AppData, usize), _env| {
+                                // We have access to both the list item's data and the app data.
                                 // Remove element from right list.
-                                shared.retain(|v| v != item);
+                                shared.del_right(*item);
                             })
                             .fix_size(80.0, 20.0)
                             .align_vertical(UnitPoint::CENTER),
                     )
                     .padding(10.0)
                     .background(Color::rgb(0.5, 0.0, 0.5))
                     .fix_height(50.0)
-            })
-            .with_spacing(10.),
-        )
+            }
+            Either::new(
```

Comment:
It might be possible to move the Either down the widget tree.

Reply:
It seems fine where it is to me. I think if I were writing it, I'd just have a single `Label` and switch the color by changing `LABEL_COLOR` in an `EnvScope`, but this seems fine too :shrug:

Reply:
I'll have a think about this before I merge.